lib/cli/transaction.rs

```diff
@@ -325,3 +332,49 @@ pub fn make_transaction_builder(
         }
     }
 }
+
+/// Retrieves a `SecretKey` based on the provided secret key string and configuration options.
+///
+/// # Arguments
+///
+/// * `secret_key` - A string representing the secret key. If empty, a `None` option is returned.
+/// * `allow_unsigned_deploy` - A boolean indicating whether unsigned deploys are allowed.
+///
+/// # Returns
+///
+/// Returns a `Result` containing an `Option<SecretKey>`. If a valid secret key is provided and the `sdk` feature is enabled,
+/// the `Result` contains `Some(SecretKey)`. If the `sdk` feature is disabled, the `Result` contains `Some(SecretKey)` parsed from the provided file.
+/// If `secret_key` is empty and `allow_unsigned_deploy` is `true`, the `Result` contains `None`. If `secret_key` is empty and `allow_unsigned_deploy` is `false`,
+/// an `Err` variant with a `CliError::InvalidArgument` is returned.
+///
+/// # Errors
+///
+/// Returns an `Err` variant with a `CliError::Core` or `CliError::InvalidArgument` if there are issues with parsing the secret key.
+pub fn get_maybe_secret_key(
+    secret_key: &str,
+    allow_unsigned_deploy: bool,
+    context: &'static str,
+) -> Result<Option<SecretKey>, CliError> {
+    if !secret_key.is_empty() {
+        #[cfg(feature = "std-fs-io")]
+        {
+            Ok(Some(parse::secret_key_from_file(secret_key)?))
+        }
+        #[cfg(not(feature = "std-fs-io"))]
+        {
+            let secret_key = SecretKey::from_pem(secret_key)
+                .map_err(|error| CliError::Core(crate::Error::CryptoError { context, error }))?;
+            Ok(Some(secret_key))
+        }
+    } else if !allow_unsigned_deploy {
+        Err(CliError::InvalidArgument {
+            context,
+            error: format!(
+                "allow_unsigned_deploy was {}, but no secret key was provided",
+                allow_unsigned_deploy
+            ),
+        })
+    } else {
+        Ok(None)
+    }
```

Comment:
A nit, but it might be more legible if we use `match (secret_key.is_empty(), allow_unsigned_deploy) {..}` statement here to avoid `if/else if/else` tree.

Reply:
This commit should follow your reco 

https://github.com/casper-ecosystem/casper-client-rs/pull/153/commits/49cf58939cb148def454b9ebe5a67e1730e7e9b6